atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/OracleTableNameUnmapper.java

```diff
@@ -63,6 +66,28 @@ public String getShortTableNameFromMappingTable(
         }
     }
 
+    public Set<String> getLongTableNamesFromMappingTable(
+            ConnectionSupplier connectionSupplier, Set<String> shortTableNames) throws TableMappingNotFoundException {
+        if (shortTableNames.isEmpty()) {
+            return Set.of();
+        }
+
+        String placeHolders = String.join(",", Collections.nCopies(shortTableNames.size(), "?"));
+
+        SqlConnection conn = connectionSupplier.get();
+        AgnosticResultSet results = conn.selectResultSetUnregisteredQuery(
+                "SELECT table_name FROM " + AtlasDbConstants.ORACLE_NAME_MAPPING_TABLE + " WHERE"
```

Comment:
Relevant points:
1. Dynamic generation of the prepared statements - You can't shove an array into a single IN (?), so you either inline the array with String join(,) or use prepared statements. I went with prepared statements to avoid needing to trust the data entirely, with the small cost of needing to prepare a statement that will only be used once, but only needs one request. The alternative was to say, batch in groups of 3 and use a single prepared statement, but that will require multiple queries for perhaps negative gain.

2. For reasons described in the point above - these methods work on sets of table names, whereas other methods in the class work on single table names. I went with a batching method since we're unmapping multiple tables at a time, so there's no reason to be making 20+ queries.

3. Oracle stores table data in upper case unless explicitly asked not to at runtime. However, we store internal metadata and mapping in user provided case, which means we need to do case insensitive matching. The easiest way to achieve this is by lower casing (or upper casing) both, but naturally this is locale dependent. We could compare with both lower == lower OR upper == upper to catch languages that have different casing, but that still doesn't provide a true case insensitive search. I'm not entirely sure we need it here? We do fail early if we can't find the mapping
4. 
5. There's no caching, since each table will only be unmapped once before deletion

Reply:
For 1: unfortunately there's a way I can see this go wrong in that the maximum number of placeholders allowed in an IN clause is limited by the definition of Oracle's flavour of SQL to 1000. I'm afraid we'll need to chunk the tables in groups of 1000. See https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/IN-Condition.html#GUID-C7961CB3-8F60-47E0-96EB-BDCF5DB1317C

For 3: this makes sense, but should be documented in the code by virtue of comment.

2 and 4 are fine.

Reply:
:/ I was going off the max query size (which is undefined for Oracle, it's configuration dependent) (https://docs.oracle.com/en/database/oracle/oracle-database/19/refrn/logical-database-limits.html#GUID-685230CF-63F5-4C5A-B8B0-037C566BDA76). I've added batching at 900 entries, which we really are very unlikely to hit (especially from the proxy side, as we're deleting all table first before doing this) but at least we have that protection.

On query size, SQLite is 1,000,000 chars and we're at 59 + metadata table name + 8 * 900 = 7259 + metadata table name, as prepared values are separate to the query size limit. Oracle's is undefined.

With the latest change, there's now a 5.

5. We push lower to the DB engine rather than doing it in the client to reduce locale and implementation mismatches. Given 1) lower is not especially expensive, 2) this isn't a ice cold path, 3) in the majority of cases, the number of tables is small I figured this was reasonable (We won't have an index on lower case variants of the key anyway since indices are case sensitive in oracle, so that will probably end up dominating.- but my point is these are all premature optimisations anyway given they're not crazy)